File: Яндекс/CodeRun/dfs.cpp
#include <iostream>
#include <vector>
#include <algorithm>

void dfs(int v, const std::vector<std::vector<int>>& matrix, std::vector<bool>& visited, std::vector<int>& ans);

int main() 
{
    int v, e;
    std::cin >> v >> e;
    
    std::vector<std::vector<int>> matrix(v + 1, std::vector<int>(v + 1, 0));
    std::vector<bool> visited(v + 1, false);
    std::vector<int> ans;

    for(int i = 0; i < e; ++i){
        int f, s;
        std::cin >> f >> s;
        matrix[f][s] = 1;
        matrix[s][f] = 1;
    }
    dfs(1, matrix, visited, ans);
    std::sort(ans.begin(), ans.end());
    std::cout << ans.size() << '\n';
    for(auto& a : ans) std::cout << a << " ";
    
    return 0;
}

void dfs(int v, const std::vector<std::vector<int>>& matrix, std::vector<bool>& visited, std::vector<int>& ans) {
    visited[v] = true;
    ans.push_back(v);
    for (int u = 1; u < matrix[v].size(); ++u) {
        if (!visited[u] && matrix[v][u] == 1) {
            dfs(u, matrix, visited, ans);
        }
    }
}

File: Coderun and Leetcode/Coderun/DFS.go
package main

import (
    "bufio"
    "fmt"
    "sort"
    "os"
)

func dfs(v int, matrix [][]int, visited []bool, ans []int) []int {
    visited[v] = true
    ans = append(ans, v)
    for u := 1; u < len(matrix[v]); u++ {
        if !visited[u] && matrix[v][u] == 1 {
            ans = dfs(u, matrix, visited, ans)
        }
    }
    return ans
}

func main() {
    reader := bufio.NewReader(os.Stdin)
    writer := bufio.NewWriter(os.Stdout)
    defer writer.Flush()

    var n, m int
    fmt.Fscan(reader, &n, &m)

    matrix := make([][]int, n+1)
    for i := 1; i <= n; i++ {
        matrix[i] = make([]int, n+1)
    }

    visited := make([]bool, n+1)
    ans := make([]int, 0)

    for i := 0; i < m; i++ {
        var f, s int
        fmt.Fscan(reader, &f, &s)
        matrix[f][s] = 1
        matrix[s][f] = 1
    }

    ans = dfs(1, matrix, visited, ans)
    sort.Ints(ans)
    fmt.Fprintln(writer, len(ans))
    for _, val := range ans {
        fmt.Fprintf(writer, "%d ", val)
    }
}